tests/Benchmarks/System.Text.Encodings.Web.Utf8/UrlEncoderPerf.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using System.Linq;
+
+namespace System.Text.Encodings.Web.Utf8.Benchmarks
+{
+    public class UrlEncoderPerf
+    {
+        private byte[] _encodedBytes;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            var random = new Random(42);
+            var builder = new StringBuilder();
+            for (int i = 0; i < 10000; i++)
+            {
+                if (random.NextDouble() < 0.15)
```

Comment:
Why did you decide on 0.15 here? Out of curiosity, can you build the string to be all hex characters (locally) and share the results? I want to try and exaggerate the use of ReadHex (before) vs Utf8Parser.TryParse (after) to see if the difference grows.

Otherwise, this change looks good to me. Thanks!

cc @adamsitnik

Reply:
I wanted something close to real life data so I picked some ratio that looks reasonable. I can change it if you have some better guesstimate.

The results for all hex chars:

Before:

|        Method |     Mean |    Error |   StdDev |   Median |
|-------------- |---------:|---------:|---------:|---------:|
| DecodeInPlace | 262.7 us | 12.84 us | 37.66 us | 245.6 us |

After:

|        Method |     Mean |    Error |   StdDev |
|-------------- |---------:|---------:|---------:|
| DecodeInPlace | 234.7 us | 4.680 us | 6.085 us |

:)
I'm not sure how consistent these tests are though.

Reply:
It's ok to use Random for input data, if the seed and length are constant. (so we always have the same input and compare apples to apples between the runs)